integration_tests/test_git_source.py

```diff
@@ -142,3 +165,47 @@ def test_pull_git_with_depth(self):
             stdout=subprocess.DEVNULL)
 
         self.run_snapcraft('pull')
+
+
+class GitGenerateVersionTestCase(GitSourceBaseTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.init_and_config_git()
+        os.mkdir('snap')
+
+        with open(os.path.join('snap', 'snapcraft.yaml'), 'w') as f:
+            print(dedent("""\
+                name: git-test
+                version: git
+                summary: test git generated version
+                description: test git generated version with git hint
+                architectures: [amd64]
+                parts:
+                    nil:
+                        plugin: nil
+                """), file=f)
```

Comment:
Almost all the other tests use an existing yaml instead of writing one. Is there a reason here to do it differently?